pkg/apisix/resource_test.go

```diff
@@ -40,7 +40,10 @@ func TestItemUnmarshalJSON(t *testing.T) {
 }
 `
 	err = json.Unmarshal([]byte(emptyData), &items)
-	assert.Equal(t, err.Error(), "unexpected non-empty object")
+	assert.True(t,
```

Comment:
Could you give your reasons for these changes? It's acceptable throwing some exceptions in the tests.

Reply:
When err is nil, will trigger panic. Add nil value judgment to avoid panic.